R/ci.R

```diff
@@ -116,51 +138,72 @@ use_codecov_badge <- function() {
 }
 
 use_coveralls_badge <- function() {
-  check_uses_github()
+  uses_github()
   url <- glue("https://coveralls.io/r/{github_repo_spec()}?branch=master")
   img <- glue(
     "https://coveralls.io/repos/github/{github_repo_spec()}/badge.svg"
   )
   use_badge("Coverage status", url, img)
 }
 
-#' @section `use_appveyor()`:
-#' Adds a basic `appveyor.yml` to the top-level directory of a package. This is
-#' a configuration file for the [AppVeyor](https://www.appveyor.com) continuous
-#' integration service for Windows.
+#' Setup deployment for Travis CI
+#'
+#' Creates a public-private key pair,
+#' adds the public key to the GitHub repository via [github_add_key()],
+#' and stores the private key as an encrypted environment variable in Travis CI
+#' via [travis_set_var()],
+#' possibly in a different repository.
+#' The \pkg{tic} companion package contains facilities for installing such a key
+#' during a Travis CI build.
+#'
+#' @importFrom travis travis_set_var
+#' @importFrom travis github_add_key
+#' @importFrom openssl rsa_keygen
+#' @param path `[string]`\cr
+#'   The path to a GitHub-enabled Git repository (or a subdirectory thereof).
+#' @param info `[list]`\cr
+#'   GitHub information for the repository, by default obtained through
+#'   [github_info()].
+#' @param repo `[string|numeric]`\cr
+#'   The GitHub repo slug, by default obtained through [github_repo()].
+#'   Alternatively, the Travis CI repo ID, e.g. obtained through `travis_repo_id()`.
+#' @param travis_repo `[string]`\cr
+#'   The Travis CI repository to add the private key to, default: `repo`
+#'   (the GitHub repo to which the public deploy key is added).
+#'
 #' @export
-#' @rdname ci
-use_appveyor <- function(browse = interactive()) {
-  check_uses_github()
+use_travis_deploy <- function(path = ".", info = travis:::github_info(path),
+                              repo = travis:::github_repo(info = info),
+                              travis_repo = repo) {
 
-  use_template("appveyor.yml", ignore = TRUE)
+  # authenticate on github and travis and set up keys/vars
 
-  appveyor_activate(browse)
-  use_appveyor_badge()
+  # generate deploy key pair
+  key <- rsa_keygen()  # TOOD: num bits?
 
-  invisible(TRUE)
-}
+  # encrypt private key using tempkey and iv
+  pub_key <- travis:::get_public_key(key)
+  private_key <- travis:::encode_private_key(key)
```

Comment:
These two functions should now better live in _tic_ (https://github.com/ropenscilabs/tic/issues/67).

Reply:
exported or not?

Reply:
Exported, the `:::` is caught by the CRAN checks (and effectively these functions are part of _tic_'s API).

Reply:
See respective PRs.